Add docs for circuit batching (#3134)
quantumlib/Cirq: docs/google/best_practices.md

that your parameter sweeps, especially products of sweeps, do not become so excessively large
that they overcome this limit.

## Use batches if sweeps are not possible

The engine has a method called `run_batch()` that can be used to send multiple
circuits in a single request.  This can be used to increase the efficiency

> and rep rate of your program.

Reviewer: repetition rate? Maybe we want to define that concept too?

Author: Done.

In order to efficiently batch circuits, the circuits in these batches must
measure the same qubits and have the same number of repetitions for each

> circuit.

Reviewer: What happens when there is a difference between measurements? How bad is it going to be? 
What amount of repetition difference are we tolerating? Can you explain a little bit more why these constraints result in the most efficient batching?

Author: I added a short note onto this.  However, the reasons for this are fairly deep in the hardware and I don't think it's appropriate to put too much more information on the public page.

Reviewer: That sounds good - that was what I was looking for!

that your parameter sweeps, especially products of sweeps, do not become so excessively large
that they overcome this limit.

## Use batches if sweeps are not possible

The engine has a method called `run_batch()` that can be used to send multiple
circuits in a single request.  This can be used to increase the efficiency
of your program so that more repetitions are completed per second.

> In order to efficiently batch circuits, the circuits in these batches must

Reviewer: ```suggestion
The circuits that are meant to be executed in batches must
```
I.e. there is no such thing as "efficiently batching" and "less efficiently batching" - it seems that this is more binary - you must measure the same qubits otherwise there is no batching.

measure the same qubits and have the same number of repetitions for each

> circuit.  Otherwise, the circuits will not be able to batched together

Reviewer: ```suggestion
circuit.  Otherwise, the circuits will not be batched together
```
